crates/apub_lib/src/object_id.rs

```diff
@@ -107,6 +97,7 @@ where
   async fn dereference_from_http(
     &self,
     data: &<Kind as ApubObject>::DataType,
+    client: &ClientWithMiddleware,
```

Comment:
Thanks, this is much better.

Reply:
A bit verbose. I liked how it worked before making the data generic, then the client could simply be passed there. In fact that should also work with generics, but not sure if its a good idea.